prboom2/src/d_deh.c

```diff
@@ -3522,3 +3513,47 @@ dboolean deh_GetData(char *s, char *k, uint_64_t *l, char **strval, FILE *fpout)
 
   return(okrc);
 }
+
+//
+// CheckDehConsistency
+//
+void CheckDehConsistency(void)
+{
+  int i, j, maxargs;
+  dboolean found;
+  const char *bexptr_name;
+
+  // sanity-check bfgcells and bfg ammopershot
+  if (
+    bfgcells_modified &&
+    weaponinfo[MT_BFG].intflags & WIF_ENABLEAPS &&
+    bfgcells != weaponinfo[MT_BFG].ammopershot
+  )
+    I_Error("Mismatch between bfgcells and bfg ammo per shot modifications! Check your dehacked.");
+
+  // ensure states don't use more mbf21 args than their
+  // action pointer expects, for future-proofing's sake
+  for (i = 0; i < num_states; i++)
+  {
+    bexptr_name = "(NULL)";
+    maxargs = 0;
+    found = FALSE;
+    j= -1;
+
+    do
+    {
+      ++j;
+      if (states[i].action == deh_bexptrs[j].cptr)
+      {
+        bexptr_name = deh_bexptrs[j].lookup;
+        maxargs = deh_bexptrs[j].argcount;
+        found = TRUE;
+      }
+    } while (!found && (deh_bexptrs[j].cptr != NULL));
+
```

Comment:
Don't we need an "if not found, continue" here?

Reply:
Not really -- if there's no matching codepointer, we want to make sure there are zero args defined. Probably doesn't matter in practice for null-action states, but it's nice n' consistent.

That said, in practice, states with no codepointer match the A_NULL entry, so there's never not a match. :P

Reply:
Not really -- if there's no matching codepointer, we want to make sure there are zero args defined. Probably doesn't matter in practice for null-action states, but it's nice n' consistent.

That said, in practice, states with no codepointer match the A_NULL entry, so there's never not a match. :P

Reply:
Oh, but then why do you have a found variable? 🤔 That's what was throwing me off. I.e., just `break` when you find something?

Reply:
Also, is something stopping this from just being `for (j = 0; deh_bexptrs[j].cptr != NULL; ++j)`? The `do while` doesn't feel natural here to me :^)

Reply:
Changed to a `break`. A classic case of "not enough coffee" :P

Reply:
I updated the loop too 🧐 
If there's anything wrong let me know and we'll fix it - merging for now.

Reply:
Nah, is good -- the do/while was me copying the old code style, but I'm all for de-Killough-izing the new bits if you are. :P